Feature/basic auth implementation

ankita-turing opened the pull request:
Fixed 54 but withg a bug. That's why it was replaced with the PR # 83

ankita-turing committed:
content manager: new feature added basic auth implementation

# Conflicts:
#	src/test/java/com/dehold/contentmanager/user/web/UserControllerIntegrationTest.java

ankita-turing committed:
content manager: updated test cases for username password auth in UserServiceTest.java

ankita-turing committed:
content manager: updated test cases for basic auth implementation

ankita-turing committed:
content manager: csrf implementation added for only tests for basic auth implementation

ankita-turing committed:
content manager: csrf config changed for only tests for basic auth implementation

dholde reviewed (changes requested):
Hi @ankita-turing ,

Here are some improvements to be introduced:

1. Test to issue alignment: The test `createUser_shouldEncodePasswordBeforeSaving` assumes that the service will encrypt the password. This is not stated in the issue description (could happen in the repository as well). -> Either mention that the service method that you test will encrypt the pw or test the encryption in an integration test (Like create a new test case in the `UserControllerIntegrationTest`).
2. Test to issue alignment: The test case `getUser_shouldThrowEntityNotFoundExceptionWhenUserDoesNotExist` asserts on a specific exception which is not mentioned in the issue description.
3. Test to issue alignment: The test case `updateUser_shouldUpdateExistingUserFieldsProperly` checks for updating the username and password. This is not mentioned in the issue description.
4. Issue clarity: The issue description does not mention any security configuration, but it's in the code `src/main/java/com/dehold/contentmanager/config/SecurityConfig.java`. Either remove the configuration or mention explicitly that a dummy configuration allowing all requests unauthenticated should be added. (I would recommend to remove the config).

ankita-turing committed:
content manager: removed irrelevant test case from basic auth implementation

dholde reviewed (changes requested):
Hi @ankita-turing ,

The test-to-issue-alignment would not be sufficient due to false negatives. For example there is an assertion
`assertEquals("The entity User with id " + randomId + " does not exist", exception.getMessage());` that checks for a specific error message which is not mentioned in the issue description. 

Please make sure to have only tested what is mentioned in the issue description (also for potential other functionality that you explicitly verify in the tests).

ankita-turing committed:
content manager: added correct test case for basic auth

dholde committed:
Proper test profile loading + security config

Load test profile via application.yml. Also provide specific test security config

ankita-turing committed:
content manager: corrected BlogPostControllerIntegrationTest setup method

dholde committed:
Fix BlogPost test setup

dholde committed:
Merge branch 'feature/basic-auth-implementation' of github.com:Turing-dev-community/content-manager into feature/basic-auth-implementation
